main.tf

```diff
@@ -0,0 +1,176 @@
+  service_principal_id = azuread_service_principal.sp.id
+  value                = random_password.aks.result
+  end_date             = "2099-01-01T01:00:00Z"
+}
+
+resource "azurerm_role_assignment" "sp" {
+  scope                = data.azurerm_subscription.current.id
+  role_definition_name = "Owner"
+  principal_id         = azuread_service_principal.sp.object_id
+}
+
+resource "azurerm_kubernetes_cluster" "aks" {
+  lifecycle {
+    ignore_changes = [
+      default_node_pool[0].node_count
+    ]
+  }
+
+  name                = local.cluster_name
+  location            = azurerm_resource_group.k8s.location
+  resource_group_name = azurerm_resource_group.k8s.name
+  dns_prefix          = local.cluster_name
+
+  default_node_pool {
+    name                = "platform"
+    node_count          = var.node_pools.platform.cluster_auto_scaling_max_count
+    vm_size             = var.node_pools.platform.vm_size
+    availability_zones  = var.node_pools.platform.zones
+    max_pods            = 250
+    os_disk_size_gb     = 128
+    node_taints         = var.node_pools.platform.taints
+    enable_auto_scaling = var.node_pools.platform.cluster_auto_scaling
+    min_count           = var.node_pools.platform.cluster_auto_scaling_min_count
+    max_count           = var.node_pools.platform.cluster_auto_scaling_max_count
+  }
+
+  service_principal {
+    client_id     = azuread_service_principal.sp.application_id
+    client_secret = azuread_service_principal_password.sp.value
+  }
+
+  addon_profile {
+    oms_agent {
+      enabled                    = true
+      log_analytics_workspace_id = azurerm_log_analytics_workspace.logs.id
+    }
+  }
+
+  network_profile {
+    load_balancer_sku  = "standard"
+    network_plugin     = "azure"
+    network_policy     = "calico"
+    dns_service_ip     = "10.0.0.10"
+    docker_bridge_cidr = "172.17.0.1/16"
+    service_cidr       = "10.0.0.0/16"
+  }
+
+  windows_profile {
+    admin_username = "azureuser"
+  }
+
+  tags = {
+    Environment = "Development"
+  }
+}
+
+resource "azurerm_kubernetes_cluster_node_pool" "aks" {
+  lifecycle {
+    ignore_changes = [
+      node_count
+    ]
+  }
+
+  for_each = {
+    # Create all node pools except for 'platform' because it is the AKS default
+    for key, value in var.node_pools :
+    key => value
+    if key != "platform"
+  }
+
+  kubernetes_cluster_id = azurerm_kubernetes_cluster.aks.id
+  name                  = each.key
+  node_count            = each.value.cluster_auto_scaling_max_count
+  vm_size               = each.value.vm_size
+  availability_zones    = each.value.zones
+  max_pods              = 250
+  os_disk_size_gb       = 128
+  os_type               = each.value.node_os
+  node_taints           = each.value.taints
+  enable_auto_scaling   = each.value.cluster_auto_scaling
+  min_count             = each.value.cluster_auto_scaling_min_count
+  max_count             = each.value.cluster_auto_scaling_max_count
```

Comment:
I'd be inclined to just use the same names.

Reply:
Yeah, good suggestion